catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/CachedClientPool.java

```diff
@@ -65,7 +65,18 @@ public class CachedClientPool implements ClientPool<IMetaStoreClient, TException
     this.clientPoolCache =
         Caffeine.newBuilder()
             .expireAfterAccess(evictionInterval, TimeUnit.MILLISECONDS)
-            .removalListener((ignored, value, cause) -> ((HiveClientPool) value).close())
+            .removalListener(
+                (ignored, value, cause) -> {
+                  // Set the class loader to the isolated class loader.
+                  ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+                  Thread.currentThread()
+                      .setContextClassLoader(CachedClientPool.class.getClassLoader());
```

Comment:
why this works? what's CachedClientPool classloader?

Reply:
``Java
ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
```

The classloader is `AppClassloader`,  Close Hive-related resource needs the Isolated classloader.

Reply:
It appears that AppClassloader is working fine here, the root cause is not at this point.

Reply:
I prefer to dig out the root cause to check whether there are  some problems with current Isolated classloader.

Reply:
Yep, we should have a reason to explain why.

Reply:
I have confirmed the root cause: We need to ensure that the removal listener is called before the isolated class loader is closed, as the code logic in the removal listener is asynchronous.  Please refer to the most recent changes.